spec/lib/landable/event_publisher_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'spec_helper'
+
+module Landable
+  module Traffic
+    describe 'EventPublisher', type: :controller do
+
+      controller(ApplicationController) do
+        include Landable::Traffic
+        prepend_around_action :track_with_landable!
+
+        def my_path
+          render nothing: true
+        end
+      end
+
+      before do
+        routes.draw do
+          get '/my_path' => 'anonymous#my_path'
+          post '/my_path' => 'anonymous#my_path'
+          delete '/my_path' => 'anonymous#my_path'
+        end
+      end
+
+      let(:message_keys) do
+        [:ad_group, :ad_type, :bid_match_type, :campaign, :content, :creative, :device_type,
+         :experiment, :keyword, :match_type, :medium, :network, :placement, :position,
+         :search_term, :source, :target]
+      end
+
+      let(:attribution) do
+        message_keys.each_with_object({}) do |attribute, hash|
+          hash[attribute] = "test_#{attribute}"
+        end
+      end
+
+      let(:tracker) { controller.instance_variable_get(:@tracker) }
+      let(:page_view) { PageView.last }
+      let(:published_message) { EventPublisher.new(page_view).message }
+
+      it 'should properly properly set the attribution data and send it within a message' do
+        get :my_path, attribution
+        #binding.pry
+        # TO DO
```

Comment:
the message is getting tested, just attributions is not working.  Still working on this last test... the others are working, we're 6/7 passing.